cryptarchia/messages.abnf

```diff
@@ -3,13 +3,14 @@
 BLOCK            = HEADER CONTENT
 ; ------------ HEADER ---------------------
 VERSION          = %x01
-HEADER           = VERSION HEADER-UNSIGNED
+HEADER           = VERSION HEADER-UNSIGNED LEADER-PROOF
 HEADER-UNSIGNED  = %x00 HEADER-COMMON
 HEADER-COMMON    = CONTENT-SIZE CONTENT-ID BLOCK-DATE PARENT-ID
 CONTENT-SIZE     = U32
 BLOCK-DATE       = BLOCK-SLOT
 BLOCK-SLOT       = U64
 PARENT-ID        = HEADER-ID
+LEADER-PROOF     = <NOT SPECIFIED YET>
```

Comment:
mmm, isn't the point of this to specify things? 
If we don't plan on having this ready for v1 I would remove this, as otherwise it's not a spec.

Apart from this, `HEADER-UNSIGNED` was meant to be one possible variant without leader proof. If we want to add a variant with it, I would suggest we create another rule and add it as a choice, e.g.:

```
HEADER = VERSION (HEADER-UNSIGNED / HEADER-SIGNEDv1) ; either HEADER-UNSIGNED or HEADER-SIGNEDv1 can follow 
HEADER-SIGNEDv1 = %x01 HEADER-COMMON LEADER-PROOF
```

Reply:
I thought the intention was to have HEADER-UNSIGNED be the data that is signed over and the the "sig" goes outside of that and signs over the "unsigned" data.

> mmm, isn't the point of this to specify things?

The leader proof details depend on the proof system we will use, I think there's still value in setting up the structure that will be filled in by the CL details, no? right now the leader proof is just a mock structure that skips over the cryptography bits.

Reply:
> The leader proof details depend on the proof system we will use, I think there's still value in setting up the structure that will be filled in by the CL details, no? right now the leader proof is just a mock structure that skips over the cryptography bits.

I would leave this out of the specs for v1 if the spec are meant to be a reference to implement the protocol in a compatible way. What would you do in the implementation with a spec that says "I don't know"? IMO, if we don't know how the transaction bits will look like, we better leave this off for now, but do that in a way that 1) signals it will be introduced in the future 2) sets a path for said future improvement.

The header unsigned variant was meant exactly for those two purposes, and does not need to stay in all future revisions (even though a genesis block, if we will have one, won't contain a leadership proof)

Reply:
So for the purposes of the v1 spec, we do want mock leader proofs in the headers since these leader proofs are used in the state update logic and this logic goes beyond merely proof verification. (i.e. there is leader proof importing for orphan blocks that needs to be specified)

I could specify a `MOCK-LEADER-PROOF` here that is implementable, in fact I'll do that now, and similarly to the unsigned variant, we can replace it with the real deal once we understand CL more.